### DISCUSSION TITLE
Added: LDM notes for September 5, 2018

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#1849] [General]
https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-09-05.md

# Agenda

1. Index operator: is it a unary operator?
1. Compiler intrinsics

**Comment by qrli** [comment]
The index operator feels a bit special compared to other operators. Does it has to be?
Or, can it use the same pattern as `arr?[i]`, so that it would be `arr^[i]` instead of `arr[^i]`, which would eliminate the free `^i` scenarios.

**Comment by yaakov-h** [comment]
@qrli that wouldn't let you store an index-from-the-rear into a variable. You would need a second variable to track the direction.

**Comment by qrli** [comment]
@yaakov-h But why do you want to store an index-from-the-rear, instead of an integer, into a variable.

**Comment by yaakov-h** [comment]
So that you can pass it to a function, or hang onto it from later, or return it from a calculation, or, or, or...

**Comment by 4creators** [comment]
Due to lack of reaction on my question in compiler intrinsics thread I would like to ask again: is there any chance to change milestone for compiler intrinsics from X.X to `AnyTime`?

@gafter @jaredpar @agocke @MadsTorgersen

**Comment by jaredpar** [comment]
I'm not sure what you want that change to achieve.

**Comment by agocke** [comment]
@4creators The notes say that we don't want to accept the proposal as is, so if we got an implementation of the current proposal we would reject it.

**Comment by 4creators** [comment]
My question is directed at the timing the feature would be open for contributors for implementation. My understanding is that design is not ready yet to be accepted. In case there would be any roadmap for feature it would be great to know it.

**Comment by agocke** [comment]
Until we have the proposal in a state LDM likes, I don't think we can give a milestone for implementation. As to when we'll get the next proposal iteration through LDM...? I'm not sure. @jaredpar is still working on the proposal AFAIK and the LDM schedule often comes down to when people are available to meet and whether we have high priority things to discuss, like questions about proposals we've already set in the 8.0 time frame.

**Comment by qrli** [comment]
@yaakov-h Yes, you can. But why? It does not give you any advantage over plane integer.

**Comment by TheUnlocked** [comment]
The notes say that the index operator doesn't need another overload, but wouldn't a `System.Range` overload be necessary? Perhaps I'm misunderstanding something...

**Comment by jaredpar** [comment]
> My question is directed at the timing the feature would be open for contributors for implementation.

The milestone attached to a feature does not restrict when community contributions would be accepted. It is instead a statement of what type of release we feel is appropriate for a feature. 

- X.X notation: we feel it could come in a major or minor release. 
- X.0 notation: we feel this is a feature that is to big to fit into a minor release and should be major release only.

Even so if we're in the middle of a minor release that doesn't mean we would stop community from working on a major release feature. It's just going to be less of a priority to review and it may have to be held in a side branch until a major release comes along that it could be merged into.

That being said will also echo what @agocke said that the feature still doesn't have an agreed upon design. I'm actively working on a few other proposals here for LDM to take a look at. In the mean time there are plenty of other features with solid designs that lack implementations.

**Comment by 4creators** [comment]
@jaredpar Thanks for thorough explanation.

**Comment by agocke** [comment]
@TheUnlocked You're thinking of the "indexer", while we're referring to the "index operator" (^). This is indeed confusing, which is why we're renaming '^' to the "hat operator".

**Comment by TheUnlocked** [comment]
@agocke Yeah that makes a lot more sense now.
